exporter/cortexexporter/cortex_test.go

```diff
@@ -0,0 +1,539 @@
+	ce, err := newCortexExporter(config.HTTPClientSettings.Endpoint, config.Namespace, c)
+	require.NoError(t, err)
+	require.NotNil(t, ce)
+	assert.NotNil(t, ce.namespace)
+	assert.NotNil(t, ce.endpoint)
+	assert.NotNil(t, ce.client)
+	assert.NotNil(t, ce.closeChan)
+	assert.NotNil(t, ce.wg)
+}
+
+// Bug{@huyan0} success case pass but it should fail; this is because the server gets no request because export() is
+// empty. This test cannot run until export is finished.
+// Test_pushMetrics the correctness and the number of points
+func Test_pushMetrics(t *testing.T) {
+	noTempBatch := pdatautil.MetricsFromInternalMetrics(testdata.GenerateMetricDataManyMetricsSameResource(10))
+	noDescBatch := pdatautil.MetricsFromInternalMetrics(testdata.GenerateMetricDataMetricTypeInvalid())
+	// 10 counter metrics, 2 points in each. Two TimeSeries in total
+	batch := testdata.GenerateMetricDataManyMetricsSameResource(10)
+	setCumulative(batch)
+	successBatch := pdatautil.MetricsFromInternalMetrics(batch)
+
+	tests := []struct {
+		name                 string
+		md                   *pdata.Metrics
+		reqTestFunc          func(t *testing.T, r *http.Request)
+		httpResponseCode     int
+		numDroppedTimeSeries int
+		returnErr            bool
+	}{
+		{
+			"no_desc_case",
+			&noDescBatch,
+			nil,
+			http.StatusAccepted,
+			pdatautil.MetricCount(noDescBatch),
+			true,
+		},
+		{
+			"no_temp_case",
+			&noTempBatch,
+			nil,
+			http.StatusAccepted,
+			pdatautil.MetricCount(noTempBatch),
+			true,
+		},
+		{
+			"http_error_case",
+			&noTempBatch,
+			nil,
+			http.StatusForbidden,
+			pdatautil.MetricCount(noTempBatch),
+			true,
+		},
+		{
+			"success_case",
+			&successBatch,
+			func(t *testing.T, r *http.Request) {
+				body, err := ioutil.ReadAll(r.Body)
+				if err != nil {
+					t.Fatal(err)
+				}
+				assert.Equal(t, "0.1.0", r.Header.Get("X-Prometheus-Remote-Write-Version:"))
+				assert.Equal(t, "Snappy", r.Header.Get("Content-Encoding"))
+				assert.NotNil(t, r.Header.Get("Tenant-id"))
+				buf := make([]byte, len(body))
+				snappy.Decode(buf, body)
+				wr := &prompb.WriteRequest{}
+				ok := proto.Unmarshal(buf, wr)
+				require.Nil(t, ok)
+				assert.EqualValues(t, 2, len(wr.Timeseries))
+			},
+			0,
+			0,
+			false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+				if tt.reqTestFunc != nil {
+					tt.reqTestFunc(t, r)
+				}
+				w.WriteHeader(tt.httpResponseCode)
+			}))
+			defer server.Close()
+
+			serverURL, uErr := url.Parse(server.URL)
+			assert.NoError(t, uErr)
+
+			config := createDefaultConfig().(*Config)
+			assert.NotNil(t, config)
+			config.HTTPClientSettings.Endpoint = serverURL.String()
+			c, err := config.HTTPClientSettings.ToClient()
+			assert.Nil(t, err)
+			sender, nErr := newCortexExporter(config.HTTPClientSettings.Endpoint, config.Namespace, c)
```

Comment:
So here the namespace and endpoint are in the wrong position. This should not be allowed to happen. Add check and test to newCortexExporter